packages/compass-components/src/components/interactive-popover.tsx

```diff
@@ -91,14 +95,33 @@ function InteractivePopover({
     });
   }, [setOpen]);
 
-  const onClickTrigger = useCallback(() => {
-    if (open) {
-      onClose();
-      return;
-    }
+  const onClickTrigger: React.EventHandler<
+    React.MouseEvent<HTMLButtonElement> | React.TouchEvent<HTMLButtonElement>
+  > = useCallback(
+    (
+      event:
+        | React.MouseEvent<HTMLButtonElement>
+        | React.TouchEvent<HTMLButtonElement>
```

Comment:
Can you clarify why touch event is added here to the type definition? This is passed as an `onClick` listener, both browser click events and corersponding React `onClick` callbacks only return MouseEvent instances for click event

Reply:
I can't because the component was already there, I just adapted it to my needs. Compass is not touch-enabled, so I wouldn't add it, but didn't want to change something that works, that doesn't leak outside and is tested because the typescript compiler complains.

Reply:
I mean, I can remove the support for the TouchEvent, it shouldn't affect in runtime, but probably I would do it async in another small PR and I would ask whoever did this and way.

Reply:
I'm not sure what's causing the hiccup here, this event didn't have any types before, you are adding them, the `trigger` `onClick` callback is expecting only the `MouseEvent` already, so why are we adding a `TouchEvent` where nothing really expects it and it's not correct in the first place? I'm not asking for any major refactoring of existing code, just not to add code that is not needed in the first place

Are you somehow mixing it with the `clickEventListener` below? This is a separate method, doesn't look like you are modifying it in this PR at all, and `clickEventListener` is indeed passed both to `mousedown` and `touchstart` so it's warranted to have a union type there

Reply:
You are right, it was part of a refactor, cleaning up and commiting.